lib/salus/scanners/pattern_search.rb

```diff
@@ -48,55 +49,57 @@ def run
           match['required'] ||= false
           match['message'] ||= ''
 
-          if shell_return[:exit_status].success? # hit
+          if shell_return.success? # hit
             if match['forbidden']
               failure_messages << "Forbidden pattern \"#{match['regex']}\" was found " \
                 "- #{match['message']}"
             end
 
-            hits = shell_return[:stdout].encode(
+            hits = shell_return.stdout.encode(
               "utf-8",
               invalid: :replace,
               undef: :replace
             ).split("\n")
 
             hits.each do |hit|
-              record_pattern_search_hit(
+              all_hits << {
                 regex: match['regex'],
                 forbidden: match['forbidden'],
                 required: match['required'],
                 msg: match['message'],
                 hit: hit
-              )
+              }
             end
 
-          elsif [1, 2].include? shell_return[:exit_status].exitstatus
-            if shell_return[:stderr].empty?
+          elsif [1, 2].include?(shell_return.status)
+            if shell_return.stderr.empty?
               # If there were no hits, but the pattern was required add an error message.
               if match['required']
                 failure_messages << "Required pattern \"#{match['regex']}\" was not found " \
                   "- #{match['message']}"
               end
             else
-              errors << shell_return[:stderr]
+              errors << { status: shell_return.status, stderr: shell_return.stderr }
             end
           else
             raise UnhandledExitStatusError,
-                  "Unknown exit status #{shell_return[:exit_status].exitstatus} from sift "\
+                  "Unknown exit status #{shell_return.status} from sift "\
                     "(grep alternative).\n" \
-                    "STDOUT: #{shell_return[:stdout]}\n" \
-                    "STDERR: #{shell_return[:stderr]}"
+                    "STDOUT: #{shell_return.stdout}\n" \
+                    "STDERR: #{shell_return.stderr}"
           end
         end
       end
 
-      if failure_messages.empty?
+      report_info(:hits, all_hits)
+      errors.each { |error| report_error('Call to sift failed', error) }
+      failure_messages.each { |message| report_error(message) }
```

Comment:
failure messages aren't really errors. They are more just what caused the scan to fail or succeed.